### ISSUE TITLE
Deadlock when running with logserver activated

### ISSUE THREAD

**Issue by JoanFM** [jina-ai/jina#936]
**Describe your problem**
When running with logserver=True there is a deadlock when FLow ends:

```python
    f = Flow(logserver=True).\
        add(uses='BaseExecutor', parallel=1).add(uses='BaseExecutor', parallel=2).build()
    with f:
        assert hasattr(f, '_sse_logger')
        pass
```

this will run forever

┆Issue is synchronized with this [Jira Task](https://jinaai.atlassian.net/browse/JINACORE-226) by [Unito](https://www.unito.io/learn-more)

**Comment by JoanFM** [comment]
Problem seems to appear when `Pod` is closed before `Pea` has handled the termination signal.

**Comment by JoanFM** [comment]
Closed by #935
